Sources/SwifterSwift/Dispatch/DispatchQueueExtensions.swift

```diff
@@ -41,6 +41,34 @@ public extension DispatchQueue {
 
         return DispatchQueue.getSpecific(key: key) != nil
     }
+    
+    /// SwifterSwift: Runs passed closure on main thread.
+    ///
+    /// - Parameter block: The closure to run on main thread
+    static func asyncOnMain(code block: @escaping () -> Void) {
+        DispatchQueue.main.async {
+            block()
+        }
+    }
+    
+    /// SwifterSwift: Runs passed closure on background thread.
+    ///
+    /// - Parameter block: The closure to run on background thread
+    static func asyncOnBg(code block: @escaping () -> Void) {
+        DispatchQueue.global().async {
+            block()
+        }
+    }
+    
+    /// SwifterSwift: Runs passed closure asynchronous after certain time interval
+    ///
+    /// - Parameter time: The time inverval after which the closure will run.
+    /// - Parameter block: The closure to run after certain time interval.
+    static func asyncAfter(time: TimeInterval, code block: @escaping () -> Void) {
```

Comment:
I like this idea. totally better than the `.now() + 123` syntax.

I wonder what do you think about an extension on `DispatchTime` instead?
like you have `.now()` there could be a `.wait(_:)` (or any other name such as `.after(_:)` or `.delay(_:)`)
```
extension DispatchTime {

    static func wait(_ interval: TimeInterval) -> Self { .now() + interval }
}
```

resulting in syntax as:
```
DispatchQueue.main.asyncAfter(deadline: .wait(10)) {
    // Do stuff...
}
```

Reply:
I don't really see much benefit in replacing `.now() + delay` with `.wait(delay)`. You're just replacing one function call with another. A cleaner option would actually be to extend `DispatchTime` with `ExpressibleByIntegerLiteral` and `ExpressibleByFloatLiteral`.

```Swift
public extension DispatchTime: ExpressibleByIntegerLiteral {
    public init(integerLiteral value: Int) {
        self.init(uptimeNanoseconds: (.now() + .seconds(value)).uptimeNanoseconds)
    }
}

extension DispatchTime: ExpressibleByFloatLiteral {
    public init(floatLiteral value: TimeInterval) {
        self.init(uptimeNanoseconds: (.now() + value).uptimeNanoseconds)
    }
}
```
The first problem is that it's semantically wrong.
There's also the limitation here is that you can't actually use constant values, only literals. I.e.
```Swift
DispatchQueue.main.asyncAfter(deadline: 10) { ... } // OK
DispatchQueue.main.asyncAfter(deadline: delay) { ... } // Cannot convert value of type 'Int' to expected argument type 'DispatchTime'
```
I think the currently suggested extension (`dispatchAfter(delay:...)`) is better, although I'd recommend making it more complete:
```Swift
public extension DispatchQueue {
    func asyncAfter(delay: Double,
                    qos: DispatchQoS = .unspecified,
                    flags: DispatchWorkItemFlags = [],
                    execute work: @escaping @convention(block) () -> Void) {
        asyncAfter(deadline: .now() + delay, qos: qos, flags: flags, execute: work)
    }
}
```

Reply:
I agree, having a direct `TimeInterval` is much easier to use, let’s keep it as you proposed.

I just originally thought that creating a whole new `DispatchQueue` function to fix **one of the parameters** would be a little too much. 
With hope that `DispatchTime` function would allow for more granular, composable and reusable solution, not limited to `DispatchQueue` case. but as you said it’s totally just replacing one function call to another. 🚀 Plus it’s probably only ever is used in combination of both.